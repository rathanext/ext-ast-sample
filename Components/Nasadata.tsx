import { View, Text, StyleSheet } from "react-native";
import React from "react";
import { TouchableOpacity } from "react-native-gesture-handler";

const Nasadata = ({ route, navigation }: any) => {
  console.log(route.params.data);
  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={{ marginTop: 20, marginLeft: 20 }}
        onPress={() => {
          navigation.goBack();
        }}
      >
        <Text testID="name" style={{ color: "#000" }}>
          Go Back
        </Text>
      </TouchableOpacity>
      <View
        style={{
          justifyContent: "center",
          alignItems: "center",
          marginTop: 50,
        }}
      >
        <Text style={styles.nasaText}>Name: {route.params.data.name}</Text>
        <Text style={styles.nasaText}>
          Nasa Jpl Url:
          {route.params.data.nasa_jpl_url}
        </Text>
        <Text style={styles.nasaText}>
          Nasa Potential:
          {route.params.data.is_potentially_hazardous_asteroid ? "Yes" : "No"}
          {/* {route.params.data.near_earth_objects[0]
          .is_potentially_hazardous_asteroid
          ? "True"
          : "false"} */}
        </Text>
      </View>
      {/* {route.params.data.is_potentially_hazardous_asteroid ? (
        <Text style={styles.nasaText}>
          Potential:
          {route.params.data.is_potentially_hazardous_asteroid
            ? "True"
            : "False"}
        </Text>
      ) : (
        <Text style={styles.nasaText}>
          Potential:
          {route.params.data.near_earth_objects[0]
            .is_potentially_hazardous_asteroid
            ? "True"
            : "False"}
        </Text>
      )} */}
    </View>
  );
};

export default Nasadata;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  nasaText: {
    fontSize: 24,
  },
});
