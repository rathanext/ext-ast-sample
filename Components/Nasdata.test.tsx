import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react-native";

import Nasadata from "./Nasadata";

const route = {
  params: {
    data: [
      {
        name: "Name",
        nasa_jpl_url: "https://test.com/v2",
        is_potentially_hazardous_asteroid: true,
      },
    ],
  },
};

describe("CountryDetails Screen Test", () => {
  test("Crenders snapshot as expected", () => {
    const home = renderer.create(<Nasadata route={route} />);
    let tree = home.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Check Texts", () => {
    const app = renderer.create(<Nasadata route={route} />);
    const elementList = app.root.findAllByType("Text");
    expect(elementList.length).toBe(4);
  });

  test("Check Views", () => {
    const app = renderer.create(<Nasadata route={route} />);
    const elementList = app.root.findAllByType("View");
    expect(elementList.length).toBe(3);
  });
  test("check text content", () => {
    const { getByTestId } = render(<Nasadata route={route} />);
    expect(getByTestId("name").children[0]).toBe("Go Back");
  });
  //   it("renders without crashing", () => {
  //     const home = renderer.create(<Nasadata route={route} />);
  //     expect(home).toBeTruthy();
  //   });
});
