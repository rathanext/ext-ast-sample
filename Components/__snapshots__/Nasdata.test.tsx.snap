// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDetails Screen Test Crenders snapshot as expected 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    exclusive={true}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
  >
    <View
      accessible={true}
      style={
        Object {
          "marginLeft": 20,
          "marginTop": 20,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
          }
        }
        testID="name"
      >
        Go Back
      </Text>
    </View>
  </RNGestureHandlerButton>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Name: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Nasa Jpl Url:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Nasa Potential:
      No
    </Text>
  </View>
</View>
`;
